rflx/pyrflx/pyrflx.py

```diff
@@ -18,13 +18,20 @@ def __init__(
         skip_message_verification: bool = False,
     ) -> None:
         self.__packages: Dict[str, Package] = {}
-        packages = set(str(m.package) for m in model.messages)
-        for p in packages:
-            self.__packages[p] = Package(p)
-            for m in [x for x in model.messages if str(x.package) == p]:
-                self.__packages[p][str(m.name)] = MessageValue(
-                    m, model.refinements, skip_message_verification
-                )
+        messages: Dict[Tuple[str, str], MessageValue] = {}
+
+        for m in model.messages:
+            p = str(m.package)
+            if p not in self.__packages:
+                self.__packages[p] = Package(p)
+            message = MessageValue(m, skip_verification=skip_message_verification)
+            messages[(p, str(m.name))] = message
+            self.__packages[p][str(m.name)] = message
+
+        for r in model.refinements:
+            messages[(str(r.pdu.package.name), str(r.pdu.name))].add_refinement(
+                RefinementValue(r, messages[(str(r.sdu.package.name), str(r.sdu.name))])
+            )
```

Comment:
Wouldn't it be easier to use `ID` instead of the tuple for `messages`?

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/501/commits/1db3013a9f545ac919090b163a2d1a03cb4351af.